### DISCUSSION TITLE
How to use boolean value(data type) in useFetcher hook

### DISCUSSION THREAD

**Discussion by sharukhkhanajm** [remix-run/remix#4823] [Q&A] [answered]
Please see the attached screenshot.

I am getting this error in my typescript setup.

could anyone please tell me how can I use different data types in useFetcher's data object other than the string?

<img width="836" alt="image" src="https://user-images.githubusercontent.com/47321682/206885675-ea66602e-504f-45c5-82c0-82f6be271c5f.png">

**Comment by sergiodxa** [comment, marked as answer]
You need to convert it to a string, for example with `String(!todo.isComplete)` or `todo.isComplete.toString()` or `todo.isComplete ? "true" : "false"` or any other way you can find to do it.

Then in your action, parse the string

```ts
let formData = await request.formData()
let isComplete = formData.get("isComplete") === "true" ? true : false;
```

**Reply by sharukhkhanajm** [reply]
Thanks for the quick reply @sergiodxa :)
